doit/control.py

```diff
@@ -158,7 +158,7 @@ def add_filtered_task(seq, f_name):
                 the_task = self.tasks[f_name]
                 # remaining items are other tasks not positional options
                 taskcmd = TaskParse([CmdOption(opt) for opt in the_task.params])
```

Comment:
This part here has some code duplication with `Task.init_options`.
My bad, we should refactor this so the actual change to fix this issue occurs in only one place.

Reply:
If you have a preferred place to centralize this code I can attempt. What would you recommend?

Reply:
I guess I would create a new method in `Task` class. `Task.set_options` or maybe just re-use `Task.init_options`.

Reply:
I've implemented this by tweaking `Task.init_options()`:

```python
    def init_options(self, args=None):
        """Put default values on options.

        This can be called with optional command line task arguments.

        Return value: unparsed command line task arguments.
        """
```

If `args` is provided the command line arguments are parsed and the return value is the list of remaining unparsed arguments. If `args` is not specified the return value is `None`.